src/reducers/index.js

```diff
@@ -511,17 +522,23 @@ export const resetSelectedTokenIfNeeded = (state, action) => {
   return state;
 };
 
-/*
+/**
  * Used when registering or creating tokens to update the wallet token list.
-*/
+ * @param {Record<string,string>} state.allTokens - Current list of allTokens
+ * @param {string} action.payload.uid - UID of token to add
+ */
 export const onNewTokens = (state, action) => {
-  // Add new created token to the all tokens set
-  state.allTokens.add(action.payload.uid);
+  // Convert `allTokens` to a Set to prevent duplicates
+  const allTokens = {...state.allTokens};
+  // Add new created token to the all tokens object
+  const newTokenUid = action.payload.uid;
+  allTokens[newTokenUid] = newTokenUid;
```

Comment:
Same here... I don't think reducers should de-duplicate anything

I understand that they're still pure functions, but makes it harder to maintain

Reply:
The spread is not converting to Set, it is actually an object and adding the token to the object.
The de-duplication is done implicitly if the key already exists.

Reply:
That's correct: this code is actually the expected behavior of a reducer.

Since it's very similar to the other one that was inadequate, it raised this confusion.